include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,497 @@
+    //uint32_t              maxGeometryOutputVertices;
+    //uint32_t              maxGeometryTotalOutputComponents;
+    //uint32_t              maxFragmentInputComponents;
+    //uint32_t              maxFragmentOutputAttachments;
+    //uint32_t              maxFragmentDualSrcAttachments;
+    //uint32_t              maxFragmentCombinedOutputResources;
+    uint32_t maxComputeSharedMemorySize;
+    //uint32_t              maxComputeWorkGroupCount[3];
+    //uint32_t              maxComputeWorkGroupInvocations;
+    uint32_t maxWorkgroupSize[3];
+    //uint32_t              subPixelPrecisionBits;
+    //uint32_t              subTexelPrecisionBits;
+    //uint32_t              mipmapPrecisionBits;
+    //uint32_t              maxDrawIndexedIndexValue;
+    uint32_t maxDrawIndirectCount;
+    //float                 maxSamplerLodBias;
+    float    maxSamplerAnisotropyLog2;
+    uint32_t maxViewports;
+    uint32_t maxViewportDims[2];
+    //float                 viewportBoundsRange[2];
+    //uint32_t              viewportSubPixelBits;
+    //size_t                minMemoryMapAlignment;
+    uint32_t bufferViewAlignment;
+    uint32_t UBOAlignment;
+    uint32_t SSBOAlignment;
+    //int32_t               minTexelOffset;
+    //uint32_t              maxTexelOffset;
+    //int32_t               minTexelGatherOffset;
+    //uint32_t              maxTexelGatherOffset;
+    //float                 minInterpolationOffset;
+    //float                 maxInterpolationOffset;
+    //uint32_t              subPixelInterpolationOffsetBits;
+    //uint32_t              maxFramebufferWidth;
+    //uint32_t              maxFramebufferHeight;
+    //uint32_t              maxFramebufferLayers;
+    //VkSampleCountFlags    framebufferColorSampleCounts;
+    //VkSampleCountFlags    framebufferDepthSampleCounts;
+    //VkSampleCountFlags    framebufferStencilSampleCounts;
+    //VkSampleCountFlags    framebufferNoAttachmentsSampleCounts;
+    //uint32_t              maxColorAttachments;
+    //VkSampleCountFlags    sampledImageColorSampleCounts;
+    //VkSampleCountFlags    sampledImageIntegerSampleCounts;
+    //VkSampleCountFlags    sampledImageDepthSampleCounts;
+    //VkSampleCountFlags    sampledImageStencilSampleCounts;
+    //VkSampleCountFlags    storageImageSampleCounts;
+    //uint32_t              maxSampleMaskWords;
+    //VkBool32              timestampComputeAndGraphics;
+    float    timestampPeriodInNanoSeconds; // timestampPeriod is the number of nanoseconds required for a timestamp query to be incremented by 1 (a float because vulkan reports), use core::rational in the future
+    //uint32_t              maxClipDistances;
+    //uint32_t              maxCullDistances;
+    //uint32_t              maxCombinedClipAndCullDistances;
+    //uint32_t              discreteQueuePriorities;
+    float pointSizeRange[2];
+    float lineWidthRange[2];
+    //float                 pointSizeGranularity;
+    //float                 lineWidthGranularity;
```

Comment:
Are these gonna be important when rendering CAD stuff?
> pointSizeGranularity is the granularity of supported point sizes. Not all point sizes in the range defined by pointSizeRange are supported. This limit specifies the granularity (or increment) between successive supported point sizes.

Reply:
yes, but I have no clue what are the GL/GLES equivalents

Reply:
ah
```
GL_POINT_SIZE_GRANULARITY
gl_line_width_granularity
```

Reply:
we have `GL_POINT_SIZE_GRANULARITY` and `GL_LINE_WIDTH_GRANULARITY` in `glcorearb.h`